PHP/PHP Source.sublime-syntax

```diff
@@ -2578,14 +2578,14 @@ contexts:
         2: meta.regex.modifier.php
         3: punctuation.definition.string.end.php
       pop: 1
-    - match: \"
+    - match: /|"
       fail: branch-point-double-quoted
     - match: ''
       push: scope:source.regexp.php
       with_prototype:
-        - match: (?=/{{regex_modifier}}"|")
+        - match: (?=/|")
           pop: 1
-        - match: \s((#)[^"]*$\n?)
+        - match: ((#)[^"]*(?:$\n?|(?=/)))
           captures:
             1: comment.regexp.php
             2: punctuation.definition.comment.regexp.php
```

Comment:
If comments don't need to be separated by space, we can probably drop one capture group.

Consuming up to `/` can also be simplified somewhat by adding it to the negative charset.

```suggestion
        - match: (#)[^/"]*$\n?
          scope: comment.regexp.php
          captures:
            1: punctuation.definition.comment.regexp.php
```

Actually it however might on the other hand be somewhat surprising to loose regexp highlighting just due to a slash in a comment.

![grafik](https://github.com/sublimehq/Packages/assets/16542113/78813f09-3563-4bb9-8b30-416414e6c04a)

```php
<?php

$pattern = '/
(?x)
\d+# comm/ent
/x';
```

Reply:
Hmm... interesting. The comment can't have unescaped `/` either.

Reply:
Pattern simplification also applies to single quoted patterns.

Reply:
It turns out the suggested change doesn't pass the following test.

```php
$regex = '/
    c{3,6}#this is comment/ux';
//        ^^^^^^^^^^^^^^^^ comment.regexp
```